### ISSUE TITLE
[BUG] - lighttpd_1.4.42-2_mipselfs does not start on tomato USB

### ISSUE THREAD

**Issue by graysky2** [Entware-ng/Entware-ng#491]
System: MIPS (Asus RT-N66U))
Firmware: Tomato USB (Toastman 1.28.0510.7)

Lighttpd will not start for me using the out-of-the-box config files.
```
# /opt/etc/init.d/S80lighttpd check
 Checking lighttpd...              dead. 

# /opt/etc/init.d/S80lighttpd start
 Starting lighttpd...              done.
```

I do not see any lighttpd running in the output of `ps | grep lighttpd`

Running lighttpd directly from the shell gives only this:
```
# lighttpd -D -f /opt/etc/lighttpd/lighttpd.conf
cat: can't open '/opt/etc/lighttpd/conf.d/*.conf': No such file or directory
2017-01-01 08:55:17: (network.c.418) socket failed: Invalid argument
```

The 2nd line is expected since I have nothing in `/opt/etc/lighttpd/conf.d/`.  Advise is welcomed.

**Comment by ryzhovau** [comment]
> cat: can't open '/opt/etc/lighttpd/conf.d/*.conf': No such file or directory

This is just a warning, you can ignore it.

> 2017-01-01 08:55:17: (network.c.418) socket failed: Invalid argument

Looks like some kernel feature missing in Tomato and there's nothing I can do about that. I can't reproduce that on Padavan's firmware.

Still, I've updated lighty to the last version because it was suffering from segfaults on `mod_cgi` mod. Give it a try.

**Comment by graysky2** [comment]
@ryzhovau - On my system, it throws that message and does not run:
```
 # /opt/etc/init.d/S80lighttpd start
 Starting lighttpd...              done.

# ps | grep lighttpd
 6071 root      1572 S    grep lighttpd
```

Perhaps premature to close the ticket?

Version: lighttpd_1.4.44-1_mipselsf.ipk.

**Comment by zyxmon** [comment]
We hope we've found the reason. Your kernel does те support CLOEXEC  and some other flags, that the latest lighttpd uses as seen from network.c.418. We will try to fix it.

**Comment by zyxmon** [comment]
fixed in 1.4.44-1a - https://github.com/Entware-ng/entware-packages/commit/eb2787c60079ae26eca69fbc974b302eea97aafd
binaries will be available soon.
